Environment_Engine/Modify/SetPanelType.cs

```diff
@@ -119,5 +119,111 @@ public static List<Panel> SetPanelType(this List<Panel> panels, PanelType type)
 
             return panels;
         }
+
+        [Description("Returns the floor panels of a space represented by Environment Panels")]
+        [Input("panelsAsSpace", "A collection of Environment Panels that represent a closed space")]
+        [Output("floorPanels", "BHoM Environment panel representing the floor of the space")]
+        public static List<Panel> FloorPanels(this List<Panel> panelsAsSpace)
+        {
+            //Find the panel(s) that are at the lowest point of the space...
+            double minZ = 1e10;
+            foreach (Panel panel in panelsAsSpace)
+            {
+                if (panel.MinimumLevel() == panel.MaximumLevel())
+                    minZ = Math.Min(minZ, panel.MinimumLevel());
+            }
+
+            List<Panel> floorPanels = panelsAsSpace.Where(x => x.MinimumLevel() == minZ && x.MaximumLevel() == minZ).ToList();
+
+            if (floorPanels.Count == 0)
+            {
+                BH.Engine.Reflection.Compute.RecordWarning("Could not find floor panel");
+                return null;
+            }
+
+            foreach (Panel panel in floorPanels)
+            {
+                if (panel.ConnectedSpaces.Where(x => x != "-1").ToList().Count == 1)
+                    panel.Type = PanelType.SlabOnGrade;
+                else if (panel.ConnectedSpaces.Where(x => x != "-1").ToList().Count == 2)
+                    panel.Type = PanelType.FloorInternal;
+            }
+
+
+            return floorPanels;
+        }
+
+        [Description("Returns the roof panels of a space represented by Environment Panels")]
+        [Input("panelsAsSpace", "A collection of Environment Panels that represent a closed space")]
+        [Output("roofPanels", "BHoM Environment panel representing the roof of the space")]
+        public static List<Panel> RoofPanels(this List<Panel> panelsAsSpace)
```

Comment:
Now the method is moved, can we call it `SetRoofPanels` instead - just a last thought I've had

Reply:
done